src/material-experimental/mdc-slider/slider.ts

```diff
@@ -872,11 +893,34 @@ class SliderAdapter implements MDCSliderAdapter {
   }
   registerInputEventHandler = <K extends EventType>
     (thumbPosition: Thumb, evtType: K, handler: SpecificEventListener<K>): void => {
-      this._delegate._getInputElement(thumbPosition).addEventListener(evtType, handler);
+      if (evtType === 'change') {
+        this.changeSubscription = this._delegate._globalChangeListener.listen((event: Event) => {
+          // We block all real slider input change events and emit fake change events from
+          // #emitChangeEvent, instead. We do this because the mdc implementation of the slider
+          // does not trigger real change events on pointer up (only on left or right arrow key
+          // down).
+          //
+          // By stopping real change events from reaching users, and dispatching fake change events
+          // (which we allow to reach the user) the slider inputs change events are triggered at
+          // the appropriate times. This allows users to listen for change events directly on the
+          // slider input as they would with a native range input.
+          if (event.target === this._delegate._getInputElement(thumbPosition)) {
+            if ((event as any).isFake) { return; }
+            event.stopImmediatePropagation();
```

Comment:
Shouldn't we use `stopPropagation` here? `stopImmediatePropagation` will also prevent any other listeners on the same element.

Reply:
We want to stop any events not dispatched by us from reaching developers, so I believe we do want to use `stopImmediatePropagation` in this case

Reply:
I'm not sure that we can realistically do that. `stopImmediatePropagation` only prevents events that are registered after the Angular Material event was registered so if somebody managed to bind an event before that, we won't be able to prevent it. IMO this is something we should tackle with documentation, rather than fighting the native behavior and dispatching fake events instead.

Reply:
We can't do it completely, but this is the earliest we can intercept the events and stop them. I will add this to today's meeting so we can have an open discussion with the rest of the team on whether or not we want to go with this solution or as you said, tackle it with documentation

Reply:
As per our discussion in the team meeting, we agreed I'd work on seeing if we could use the native inputs under the hood to trigger mouse events, and if that didn't work we'd implement the solution here to make the api as intuitive to developers as possible.

I worked on trying to use the native inputs so that mouse events were properly triggered, but it was not possible - see my comment below for more details - so we're going through with blocking these events & dispatching fake ones in their place